tests/test_oracle.py

```diff
@@ -0,0 +1,146 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from contextlib import contextmanager
+import os
+import os.path
+import secrets
+import unittest
+
+import sqlalchemy
+
+from lsst.daf.butler.registry.databases.oracle import OracleDatabase
+from lsst.daf.butler.registry import ddl
+from lsst.daf.butler.registry.tests import DatabaseTests
+
+ENVVAR = "DAF_BUTLER_ORACLE_TEST_URI"
+TEST_URI = os.environ.get(ENVVAR)
+
+
+@unittest.skipUnless(TEST_URI is not None, f"{ENVVAR} environment variable not set.")
+class OracleDatabaseTestCase(unittest.TestCase, DatabaseTests):
+
+    @classmethod
+    def setUpClass(cls):
+        # Prefix will start with this random string to allow concurrent tests.
+        cls._token = secrets.token_hex(4).lower()
+        # Next part of the prefix is this counter, which increments in setUp.
+        cls._testCounter = 1
+        # Create a single engine for all Database instances we create, to avoid
+        # repeatedly spending time connecting.
+        cls._engine = sqlalchemy.create_engine(TEST_URI, pool_size=1)
+        cls._prefixes = []
+
+    @classmethod
+    def tearDownClass(cls):
+        commands = []
+        dbapi = cls._engine.raw_connection()
+        cursor = dbapi.cursor()
+        for objectType, objectName in cursor.execute("SELECT object_type, object_name FROM user_objects"):
+            if not any(objectName.lower().startswith(prefix) for prefix in cls._prefixes):
+                continue
+            if objectType == "TABLE":
+                commands.append(f'DROP TABLE "{objectName}" CASCADE CONSTRAINTS')
+            elif objectType in ("VIEW", "PROCEDURE", "SEQUENCE"):
+                commands.append(f'DROP {objectType} "{objectName}"')
+        for command in commands:
+            cursor.execute(command)
+        cursor.close()
+
+    def setUp(self):
+        cls = type(self)
+        cls._testCounter += 1
+        self._dbCounter = 1
+
+    def makeEmptyDatabase(self, origin: int = 0) -> OracleDatabase:
+        prefix = f"test_{self._token}_{self._testCounter}_{self._dbCounter}_"
+        self._dbCounter += 1
```

Comment:
Same here, I think there is a chance for collisions depending on how pytest runs. Maybe some random string would work better here?